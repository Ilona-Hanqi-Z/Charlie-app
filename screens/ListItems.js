import React,{Component} from 'react';
import {Card, CardItem, Thumbnail,Body,Left, Right, Button, Icon} from 'native-base';
import FontAwesome from 'react-native-vector-icons/FontAwesome';
import { Text, StyleSheet,Image, Alert} from 'react-native';


class ListItems extends Component{

        constructor(props) {
            super(props);
            this.data = props.data;
            this.state = {
                isLiked:false,
                likes: this.data.runtime,
                retweets: this.data.id,
                retweet:false
            }
        }
        LikedFunction =()=> {
            this.setState({
                isLiked: !this.state.isLiked
               
            });
           
        }
        RetweetedFunction = () => {
            this.setState({

                retweet: !this.state.retweet
            });
        }
        
        render() {
            return(
             <Card>
                <CardItem>
                <Left>
                     <Thumbnail source={{uri: this.data.small_cover_image != null ?
                     this.data.small_cover_image : "https://yts.mx/assets/images/movies/el_topo_1970/small-cover.jpg"}}/>{/*user image*/}
                    <Body>
                    
                         <Text>{this.data.title_english}</Text>
                         <Text note>{this.data.year}</Text>
                        
                    </Body>
                   
                </Left>
              </CardItem>
              <CardItem>
                  <Image source={{uri: this.data.medium_cover_image != null ?
                  this.data.medium_cover_image : "https://yts.mx/assets/images/movies/el_topo_1970/small-cover.jpg"}} 
                  style={{height:200,width:null,flex:1}}/>
              </CardItem>
              <CardItem>
                  <Body>
                      <Text>
                      {this.data.summary}
                      </Text>
                  </Body>
              </CardItem>
              <CardItem style={{height:45}}>
                <Left>
                    <Button transparent onPress ={this.LikedFunction}>
                        <FontAwesome name= {this.state.isLiked ? "heart": "heart-o"}/>
<<<<<<< HEAD
                  <Text>{this.state.isLiked ? (this.state.likes+1) : this.state.likes }</Text>
                    </Button>
                    <Button transparent onPress ={this.RetweetedFunction}>
                        <FontAwesome name="retweet"/>
                        <Text>{this.state.retweet ? (this.state.retweets+1) : this.state.retweets }</Text>
                    </Button>
                    <Button transparent>
                        <FontAwesome name="share"/>
                    </Button>
=======
                        <Text> {this.state.isLiked ? (this.state.likes+1) : this.state.likes }</Text>
                    </Button>

                    <Button transparent onPress ={this.RetweetedFunction}>
                        <FontAwesome name="retweet"/>
                        <Text> {this.state.retweet ? (this.state.retweets+1) : this.state.retweets }</Text>
                    </Button>

                    <Button transparent>
                        <FontAwesome name="share"/>
                    </Button>

>>>>>>> 0ea5128beacb67ac1ff8829e34c818d9a7542a03
                    <Right>
                        <Button style={styles.readMore}>
                            <Text style ={{color:"blue"}}>READ MORE</Text>
                        </Button>
                    </Right>
                </Left>
              </CardItem>
              </Card>
   
            );
        }
       



}
export default ListItems;

const styles = StyleSheet.create({
    container: {
        flex:1,
        alignItems: 'center',
        justifyContent: 'center'
    },
    readMore : {
      backgroundColor:'white'  
    }
});