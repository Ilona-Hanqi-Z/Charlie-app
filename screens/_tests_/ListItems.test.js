import React from 'react';
import { render } from 'react-native-testing-library';
import ListItems from '../ListItems';

const item = {"id":22887,"url":"https:\/\/yts.mx\/movies\/nurse-3d-2013","imdb_code":"tt1913166","title":"Nurse 3D","title_english":"Nurse 3D","title_long":"Nurse 3D (2013)","slug":"nurse-3d-2013","year":2013,"rating":4.5,"runtime":84,"genres":["Horror","Thriller"],"summary":"By day Abby Russell is a dedicated nurse, someone you wouldn't hesitate to trust your life with. But by night, her real work begins...using her smoldering sexuality she lures cheating men to their brutal deaths and exposes them for who they really are. When a younger nurse starts to suspect Abby's actions and compromises her master plan, Abby must find a way to outsmart her long enough to bring the cheater you'd least expect to justice.","description_full":"By day Abby Russell is a dedicated nurse, someone you wouldn't hesitate to trust your life with. But by night, her real work begins...using her smoldering sexuality she lures cheating men to their brutal deaths and exposes them for who they really are. When a younger nurse starts to suspect Abby's actions and compromises her master plan, Abby must find a way to outsmart her long enough to bring the cheater you'd least expect to justice.","synopsis":"By day Abby Russell is a dedicated nurse, someone you wouldn't hesitate to trust your life with. But by night, her real work begins...using her smoldering sexuality she lures cheating men to their brutal deaths and exposes them for who they really are. When a younger nurse starts to suspect Abby's actions and compromises her master plan, Abby must find a way to outsmart her long enough to bring the cheater you'd least expect to justice.","yt_trailer_code":"ipdZPDiTLME","language":"en","mpa_rating":"R","background_image":"https:\/\/yts.mx\/assets\/images\/movies\/nurse_3d_2013\/background.jpg","background_image_original":"https:\/\/yts.mx\/assets\/images\/movies\/nurse_3d_2013\/background.jpg","small_cover_image":"https:\/\/yts.mx\/assets\/images\/movies\/nurse_3d_2013\/small-cover.jpg","medium_cover_image":"https:\/\/yts.mx\/assets\/images\/movies\/nurse_3d_2013\/medium-cover.jpg","large_cover_image":"https:\/\/yts.mx\/assets\/images\/movies\/nurse_3d_2013\/large-cover.jpg","state":"ok","torrents":[{"url":"https:\/\/yts.mx\/torrent\/download\/63D5A6C3B708F583A21A65D219935EF8E79992D3","hash":"63D5A6C3B708F583A21A65D219935EF8E79992D3","quality":"3D","type":"bluray","seeds":33,"peers":8,"size":"1.55 GB","size_bytes":1664299827,"date_uploaded":"2020-10-28 04:22:18","date_uploaded_unix":1603855338}]}

test('it renders all inputs as expected', ()=> {
    const { toJSON } = render(<ListItems data={item}/>);

    expect(toJSON()).toMatchSnapshot();
})