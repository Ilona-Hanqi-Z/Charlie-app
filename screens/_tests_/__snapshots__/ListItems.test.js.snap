// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders all inputs as expected 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "borderBottomWidth": 0.5,
      "borderColor": "#ccc",
      "borderLeftWidth": 0.5,
      "borderRadius": 2,
      "borderRightWidth": 0.5,
      "borderTopWidth": 0.5,
      "elevation": 3,
      "flexWrap": "nowrap",
      "marginBottom": 5,
      "marginLeft": 2,
      "marginRight": 2,
      "marginTop": 5,
      "shadowColor": "#000",
      "shadowOffset": Object {
        "height": 2,
        "width": 0,
      },
      "shadowOpacity": 0.1,
      "shadowRadius": 1.5,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "borderRadius": 2,
        "flexDirection": "row",
        "paddingBottom": 10,
        "paddingLeft": 15,
        "paddingRight": 15,
        "paddingTop": 10,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "center",
          "flex": 1,
          "flexDirection": "row",
        }
      }
    >
      <Image
        source={
          Object {
            "uri": "https://yts.mx/assets/images/movies/nurse_3d_2013/small-cover.jpg",
          }
        }
        style={
          Object {
            "borderRadius": 28,
            "height": 56,
            "width": 56,
          }
        }
      />
      <View
        style={
          Object {
            "alignItems": null,
            "alignSelf": "center",
            "flex": 1,
            "marginLeft": 10,
          }
        }
      >
        <Text>
          Nurse 3D
        </Text>
        <Text
          note={true}
        >
          2013
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "borderRadius": 2,
        "flexDirection": "row",
        "paddingBottom": 10,
        "paddingLeft": 15,
        "paddingRight": 15,
        "paddingTop": 10,
      }
    }
  >
    <Image
      source={
        Object {
          "uri": "https://yts.mx/assets/images/movies/nurse_3d_2013/medium-cover.jpg",
        }
      }
      style={
        Object {
          "flex": 1,
          "height": 200,
          "width": null,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "borderRadius": 2,
        "flexDirection": "row",
        "paddingBottom": 10,
        "paddingLeft": 15,
        "paddingRight": 15,
        "paddingTop": 10,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "alignSelf": "stretch",
          "flex": 1,
        }
      }
    >
      <Text>
        By day Abby Russell is a dedicated nurse, someone you wouldn't hesitate to trust your life with. But by night, her real work begins...using her smoldering sexuality she lures cheating men to their brutal deaths and exposes them for who they really are. When a younger nurse starts to suspect Abby's actions and compromises her master plan, Abby must find a way to outsmart her long enough to bring the cheater you'd least expect to justice.
      </Text>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#fff",
          "borderRadius": 2,
          "flexDirection": "row",
          "paddingBottom": 10,
          "paddingLeft": 15,
          "paddingRight": 15,
          "paddingTop": 10,
        },
        Object {
          "height": 45,
        },
      ]
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "center",
          "flex": 1,
          "flexDirection": "row",
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        forwardedRef={[Function]}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "flex-start",
            "backgroundColor": "transparent",
            "borderBottomWidth": null,
            "borderColor": "#007aff",
            "borderLeftWidth": null,
            "borderRadius": 5,
            "borderRightWidth": null,
            "borderTopWidth": null,
            "borderWidth": undefined,
            "elevation": 0,
            "flexDirection": "row",
            "height": 45,
            "justifyContent": "space-between",
            "opacity": 1,
            "paddingBottom": null,
            "paddingLeft": null,
            "paddingRight": 15,
            "paddingTop": null,
            "shadowColor": null,
            "shadowOffset": null,
            "shadowOpacity": null,
            "shadowRadius": null,
          }
        }
      >
        <Text />
        <Text>
           
          84
        </Text>
      </View>
      <View
        accessible={true}
        focusable={true}
        forwardedRef={[Function]}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "flex-start",
            "backgroundColor": "transparent",
            "borderBottomWidth": null,
            "borderColor": "#007aff",
            "borderLeftWidth": null,
            "borderRadius": 5,
            "borderRightWidth": null,
            "borderTopWidth": null,
            "borderWidth": undefined,
            "elevation": 0,
            "flexDirection": "row",
            "height": 45,
            "justifyContent": "space-between",
            "opacity": 1,
            "paddingBottom": null,
            "paddingLeft": null,
            "paddingRight": 15,
            "paddingTop": null,
            "shadowColor": null,
            "shadowOffset": null,
            "shadowOpacity": null,
            "shadowRadius": null,
          }
        }
      >
        <Text />
        <Text>
           
          22887
        </Text>
      </View>
      <View
        accessible={true}
        focusable={false}
        forwardedRef={[Function]}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "flex-start",
            "backgroundColor": "transparent",
            "borderBottomWidth": null,
            "borderColor": "#007aff",
            "borderLeftWidth": null,
            "borderRadius": 5,
            "borderRightWidth": null,
            "borderTopWidth": null,
            "borderWidth": undefined,
            "elevation": 0,
            "flexDirection": "row",
            "height": 45,
            "justifyContent": "space-between",
            "opacity": 1,
            "paddingBottom": null,
            "paddingLeft": null,
            "paddingRight": 15,
            "paddingTop": null,
            "shadowColor": null,
            "shadowOffset": null,
            "shadowOpacity": null,
            "shadowRadius": null,
          }
        }
      >
        <Text />
      </View>
      <View
        style={
          Object {
            "alignItems": "flex-end",
            "alignSelf": "center",
            "flex": 1,
          }
        }
      >
        <View
          accessible={true}
          focusable={false}
          forwardedRef={[Function]}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "alignSelf": null,
              "backgroundColor": "white",
              "borderBottomWidth": null,
              "borderColor": "#007aff",
              "borderLeftWidth": null,
              "borderRadius": 5,
              "borderRightWidth": null,
              "borderTopWidth": null,
              "borderWidth": undefined,
              "elevation": 2,
              "flexDirection": "row",
              "height": 45,
              "justifyContent": "space-between",
              "opacity": 1,
              "paddingBottom": 6,
              "paddingTop": 6,
              "shadowColor": undefined,
              "shadowOffset": undefined,
              "shadowOpacity": undefined,
              "shadowRadius": undefined,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "blue",
              }
            }
          >
            READ MORE
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
